### PR TITLE
Refine the DSP API concerning OpenRTB and the Audit Log/Button

### PR DESCRIPTION
**Pull request by RomainLofaso**

### CODE REVIEW

File: mvp-spec/ad-server-implementation.md [hunk 1]
```diff
@@ -0,0 +1,785 @@
...
+the Publisher website for a given user in addition to the inventory information. 
+Those cookies contain the encrypted Prebid SSO Data. As a reminder, 
+Prebid SSO Data are Pseudonymous-Identifiers and Preferences of the user.
+
+```mermaid
+flowchart LR
+    Publisher-- Send Placements info <br /> with Prebid SSO Cookies -->AdServer[Ad Server]
+```
+
+### Step 2: Decrypt the Prebid SSO Data
```
Comment (OlivierChirouze) [inline mvp-spec/ad-server-implementation.md:89]: (Sorry if we already discussed this in another PR): I wouldn't mention encryption but simply serialization

File: mvp-spec/dsp-api.md [hunk 2]
```diff
@@ -212,15 +239,16 @@ Note that the "data" field is a simple dictionnary.
 | source          | Source object                 | The source contains all the data for identifying the DSP and verifying the Transmission.                                                                                                                                                                                                                   |
 <!--partial-end-->
 
-### Example of a Transmission Request
+Note that the signature of the source is generated based on the follwoing:
```
Comment (OlivierChirouze) [inline mvp-spec/dsp-api.md:242]: Seems like the "following" is missing?

File: mvp-spec/dsp-api.md [hunk 3]
```diff
@@ -353,10 +432,12 @@ OpenRTB bid request can contain multiple ads - named "impression" in OpenRTB
 specifications. Therefore, a bid request can have multiple Transmission
 Requests. Those Transmissions are added in the "ext" object of each "imp" (for 
 impression) object of the Bid Request. This new object dedicated to the 
-Transmission is named "prebid_sso_transmission".
+Transmission is named "prebid_sso".
```
Comment (OlivierChirouze) [inline mvp-spec/dsp-api.md:435]: We keep using "Prebid SSO" instead of "PAF" (or equivalent) in all our documents, and I think this is fine for the moment.
However here, this is a technical key used in the data model, so I believe it's important we make sure this is the right name.

And... I'm thinking it's probably not. Shouldn't it be `paf` instead?

File: mvp-spec/dsp-api.md [hunk 4]
```diff
@@ -372,64 +453,19 @@ Transmission is named "prebid_sso_transmission".
                 "pos": 0
             },
             "ext": {
-                "prebid_sso_transmission": {
+                "prebid_sso": {
```
Comment (OlivierChirouze) [inline mvp-spec/dsp-api.md:456]: Rename to `paf` ? (see previous comment)

File: mvp-spec/dsp-api.md [hunk 5]
```diff
@@ -451,11 +487,50 @@ Transmission is named "prebid_sso_transmission".
...
     },
     "user": {
-        "id": "55816b39711f9b5acf3b90e313ed29e51665623f"
+        "id": "55816b39711f9b5acf3b90e313ed29e51665623f",
+         "ext":
+         {
+            "eids": 
+            [
+                {
+                    "source": "prebid_sso",
```
Comment (OlivierChirouze) [inline mvp-spec/dsp-api.md:496]: change value to `paf`?

File: mvp-spec/dsp-api.md [hunk 6]
```diff
@@ -451,11 +487,50 @@ Transmission is named "prebid_sso_transmission".
...
+                                    "domain": "operotor0.com",
+                                    "timestamp": 1639589531,
+                                    "signature": "12345_signature"
+                                }
+                            }
+                        }
+                    ]
+                }
+            ],
+            "prebid_sso": {
```
Comment (OlivierChirouze) [inline mvp-spec/dsp-api.md:516]: remane?

File: mvp-spec/ad-server-implementation.md (hunk 1 above)
Comment (RomainLofaso) [reply mvp-spec/ad-server-implementation.md:89]: Fixed in the other PR that you mentioned. No worries.

File: mvp-spec/dsp-api.md (hunk 3 above)
Comment (RomainLofaso) [reply mvp-spec/dsp-api.md]: I have updated with `paf` in both this PR and also the one related to the Ad Server implementation.

There are still "Prebid SSO" instance because "PAF Data" seems less descriptive than "Prebid SSO Data". To be discuss to find another word for this.

File: mvp-spec/audit-log-design.md [hunk 7]
```diff
@@ -0,0 +1,257 @@
+# The Audit Log
+
+The Audit Log is the data attached to an Addressable Content so that the user
+can see how has used her/his Prebid SSO Data for generating this content. This
+document describe technical aspects in addition of
```
Comment (OlivierChirouze) [inline mvp-spec/audit-log-design.md:5]: describes

File: mvp-spec/audit-log-design.md [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+
+Once the user has clicked on the Audit Button, she/he is redirected to the
+Audit UI. The Contracting Party generates the UI based on the validations 
+of the Prebid SSO Data and the Transmission Results. The following elements must
+appear in the page:
+
+| Element                          | Details                                   |
+|----------------------------------|-------------------------------------------|
+| List of Pseudonymous-Identifiers | Each Pseudonymous-Identifier must be paired with the name of the Operator who generated it and signed it with a Reg/Green indicator expressing the validity of the signature.                                                |
+| List of the Preferences          | Each Preference must be paired with the name of the CMP who generated it and signed it with a Red/Green indicator expressing the validity of the signature.                                                             |
```
Comment (OlivierChirouze) [inline mvp-spec/audit-log-design.md:152]: The preferences are signed together so there cannot be a pair of preferences + name + indicator.

File: mvp-spec/audit-log-design.md [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+| Seed                             | The Seed is represented by the Transaction ID and a Red/Green indicator expressing the validity of the signature                                                                                                             |
+| List of Transmission Results     | The Transmission Results available in the Audit Log. Each Transmission Results is represented by the Name of the Receiver, the status of the Transmission and a Red/Green indicator expressing the validity of the signature |
```
Comment (OlivierChirouze) [inline mvp-spec/audit-log-design.md:154]: Each Transmission Result~s~

File: mvp-spec/audit-log-design.md [hunk 10]
```diff
@@ -0,0 +1,257 @@
...
+
+The design of the page is up to the Contracting Party but it must consider 
+to have a neutral UI to be used for many different Publishers in case of a DSP.
+
+For the version v0.1, there are only one possible preference
+("opt-in") and one possible pseudonymous-identifiers ("prebid_id").
```
Comment (OlivierChirouze) [inline mvp-spec/audit-log-design.md:160]: `use_browsing_for_personalization` and `paf_browser_id` (probably deserves a dedicated PR)

File: mvp-spec/audit-log-design.md [hunk 11]
```diff
@@ -0,0 +1,257 @@
...
+However the framework is designed to handle many of them. 
+
+## Source validations
+
+As we saw, for displaying correctly the Audit UI, the Contracting Party must 
+validate different sources. For each source available in the 
+Audit Log, the Contracting Party must:
+
+* Call the Identity Endpoint of the "domain" of the source to retrieve its name
```
Comment (OlivierChirouze) [inline mvp-spec/audit-log-design.md:169]: Later simply link to signatures.md

File: mvp-spec/audit-log-design.md [hunk 12]
```diff
@@ -0,0 +1,257 @@
...
+### Verify the Preferences
+
+The Audit Log contains a list of Preferences with one signature. The UTF-8 string for a specific Preference must be built as follows:
+
+<!--partial-begin { "files": [ "preferences-signature-string.txt" ], "block": "" } -->
+<!-- ⚠️ GENERATED CONTENT - DO NOT MODIFY DIRECTLY ⚠️ -->
+```
+preferences.source.domain + '\u2063' +
+preferences.source.timestamp + '\u2063' +
+prebid_id.source.signature + '\u2063' +
```
Comment (OlivierChirouze) [inline mvp-spec/audit-log-design.md:205]: Should add a comment to explain what "prebid_id" is (the identifier that has type "prebid_id")

File: mvp-spec/partials/transmission-result-table.md [hunk 13]
```diff
@@ -1,6 +1,6 @@
 | Field           | Type                          | Details                           |
 |-----------------|-------------------------------|-----------------------------------|
-| version         | Number                        | The version of the Prebid SSO used.                                                                                                                                                                                                                               |
+| version         | Number                        | The version of the PAF used.                                                                                                                                                                                                                               |
```
Comment (OlivierChirouze) [inline mvp-spec/partials/transmission-result-table.md:3]: Maybe you should update it everywhere for consistency.

**Review by OlivierChirouze — CHANGES_REQUESTED** [review]
I think the description of preferences audit log must be adapted.

File: mvp-spec/audit-log-design.md (hunk 7 above)
Comment (RomainLofaso) [reply mvp-spec/audit-log-design.md:5]: Done

File: mvp-spec/audit-log-design.md (hunk 8 above)
Comment (RomainLofaso) [reply mvp-spec/audit-log-design.md:152]: Done

File: mvp-spec/audit-log-design.md (hunk 9 above)
Comment (RomainLofaso) [reply mvp-spec/audit-log-design.md:154]: Don't get your point. Is it the "s" at the end of the "Results" ?
Fixing it.

File: mvp-spec/audit-log-design.md (hunk 10 above)
Comment (RomainLofaso) [reply mvp-spec/audit-log-design.md:160]: We already discussed this: https://github.com/criteo/addressable-network-proposals/issues/59#issuecomment-1023292707

File: mvp-spec/audit-log-design.md (hunk 12 above)
Comment (RomainLofaso) [reply mvp-spec/audit-log-design.md:205]: I tweak the signature description to make it more understable.

Comment (OlivierChirouze) [reply mvp-spec/audit-log-design.md]: Thanks, I like it!
However, it seems you've updated the main doc but not the partial (or at least, I like the "main" version better than the partial one): `identifiers[type=prebid_id]`

File: mvp-spec/dsp-implementation.md [hunk 14]
```diff
@@ -545,268 +662,26 @@ in the Audit Logs.
...
+                        "paf": {
+                            "version": 0,
+                            "receiver": "dsp1.com",
+                            "status": "success",
+                            "details": "",
+                            "source": {
+                                "domain": "dsp1.com",
+                                "timestamp": 1639589531,
+                                "signature": "12345_signature"
+                            },
```
Comment (vmihalev) [inline mvp-spec/dsp-implementation.md:676]: Hi, Team!

Could you, please, validate if i get right that all in all, transactions placement in open rtb protocol shall look as follows?
I was not able to find expicit examples for openrtb protocol in spec.

Reqeuest to ssp:
```
{
    "id": "80ce30c53c16e6ede735f123ef6e32361bfc7b22",
    "at": 1, 
    "cur": [ "USD" ],
    "imp": [
        {
            "id": "1",
            "bidfloor": 0.03,
            "banner": {
                "h": 250,
                "w": 300,
                "pos": 0
            },
            "ext": {
                "paf": {
                    "version": 0,
                    "seed": {
                        "version": 0,
                        "transaction_id": "a0651946-0f5b-482b-8cfc-eab3644d2743",
                        "publisher": "publisher.com",
                        "source": {
                            "domain": "publisher0.com",
                            "timestamp": 1639589531,
                            "signature": "12345_signature"
                        }
                    },
                    "parents": []
                }
            }
        }
    ],
```
Reqeuest to dsp:
    1) publisher->ssp transaction signed by ssp placed into "parents section"
```
{
    "id": "80ce30c53c16e6ede735f123ef6e32361bfc7b22",
    "at": 1, 
    "cur": [ "USD" ],
    "imp": [
        {
            "id": "1",
            "bidfloor": 0.03,
            "banner": {
                "h": 250,
                "w": 300,
                "pos": 0
            },
            "ext": {
                "paf": {
                    "version": 0,
                    "seed": {
                        "version": 0,
                        "transaction_id": "a0651946-0f5b-482b-8cfc-eab3644d2743",
                        "publisher": "publisher.com",
                        "source": {
                            "domain": "publisher0.com",
                            "timestamp": 1639589531,
                            "signature": "12345_signature"
                        }
                    },
                    "parents": [
                        {
                            "version": 0,
                            "receiver": "ssp1.net",
                            "status": "success",
                            "details": "",
                            "source": {
                                "domain": "ssp1.net",
                                "timestamp": 1639589531,
                                "signature": "12345_signature"
                            },
                        }
                    ]
                }
            }
        }
    ],
```
Response from dsp to ssp:
    1) ssp->dsp transaction signed by dsp
    2) no children
```
{
    "id": "1234567890",
    "bidid": "abc1123",
    "cur": "USD",
    "seatbid": [
        {
            "seat": "512",
            "bid": [
                {
                    "id": "1",
                    "impid": "1",
                    "price": 9.43,
                    "nurl": "http://adserver.com/winnotice?impid=102",
                    "iurl": "http://adserver.com/pathtosampleimage",
                    "adomain": [ "advertiserdomain.com" ],
                    "cid": "campaign111",
                    "crid": "creative112",
                    "attr": [ 1, 2, 3, 4, 5, 6, 7, 12 ],
                    "ext": {
                        "paf": {
                            "version": 0,
                            "receiver": "dsp1.com",
                            "status": "success",
                            "details": "",
                            "source": {
                                "domain": "dsp1.com",
                                "timestamp": 1639589531,
                                "signature": "12345_signature"
                            },
                            "children": []
                        }
                    }
                }
            ]
        }
    ]
}
```
Response from ssp to publisher:

ext.paf has
    1) previousely generated publisher->ssp signature
    2) dsp signature of ssp->dsp transaction in children section:
```
{
    "id": "1234567890",
    "bidid": "abc1123",
    "cur": "USD",
    "seatbid": [
        {
            "seat": "512",
            "bid": [
                {
                    "id": "1",
                    "impid": "1",
                    "price": 9.43,
                    "nurl": "http://adserver.com/winnotice?impid=102",
                    "iurl": "http://adserver.com/pathtosampleimage",
                    "adomain": [ "advertiserdomain.com" ],
                    "cid": "campaign111",
                    "crid": "creative112",
                    "attr": [ 1, 2, 3, 4, 5, 6, 7, 12 ],
                    "ext": {
                        "paf": {
                            "version": 0,
                            "receiver": "ssp1.net",
                            "status": "success",
                            "details": "",
                            "source": {
                                "domain": "ssp1.net",
                                "timestamp": 1639589531,
                                "signature": "12345_signature"
                            },
                            "children": [
                                {
                                    "version": 0,
                                    "receiver": "dsp1.com",
                                    "status": "success",
                                    "details": "",
                                    "source": {
                                        "domain": "dsp1.com",
                                        "timestamp": 1639589531,
                                        "signature": "12345_signature"
                                    },
                                }
                            ]
                        }
                    }
                }
            ]
        }
    ]
}
```

Comment (RomainLofaso) [reply mvp-spec/dsp-implementation.md:676]: Thank you, Vasilii for your question. This is exactly that! 

The docs were designed for providing implementation details by actors. However, the repo misses indeed a document with a general overview of the flow of the transmissions. Considering adding it. :-)

**Review by OlivierChirouze — APPROVED** [review]
Thanks for the update
